### ISSUE TITLE
Filter out entry for VerifyTargetingPacksHaveDoc test

### ISSUE THREAD

**Issue by mthalman** [dotnet/source-build#3660]
The VerifyTargetingPacksHaveDoc test is constantly regressing with each 6.0.1xx servicing release. Each release alternates between requiring `Microsoft.AspNetCore.App.Ref/ref/netx.y/System.Runtime.CompilerServices.Unsafe.xml` to be listed in the MissingXmlDoc.txt baseline file and not be listed. You can see the [history of the changes to this file](https://github.com/dotnet/installer/commits/release/6.0.1xx/src/SourceBuild/tarball/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/assets/baselines/MissingXmlDoc.txt) seeing that this file entry gets swapped in and out each release.

In some cases, there is differences of whether the entry needs to be contained in MissingXmlDoc.txt, depending on which stage is being run (see https://github.com/dotnet/source-build/issues/3389). This adds further complications.

This adds unnecessary churn to the release. We should either remove the test entirely or add a filter to not include the `Microsoft.AspNetCore.App.Ref/ref/netx.y/System.Runtime.CompilerServices.Unsafe.xml` in the comparison.

**Comment by MichaelSimons** [comment]
@mthalman - I think this test should be disabled immediately to unblock the 6.0 builds until a longer term decision is made.

**Comment by MichaelSimons** [comment]
In a way this test is redundant with the SDK content tests.  The SDK content tests will provide and indication of xml files missing compared to the Microsoft build.  The SDK content tests did not exist at the time this test was originally authored.  I tend to think this test should be simply removed.

**Comment by MichaelSimons** [comment]
[Triage] There may be a product issue here in why this xml doc file comes and goes.  It would be good to validate this in the Microsoft builds.  If the Microsoft builds don't have this oscillation issue then it is source build specific.  Regardless a separate issue should be logged to address this problem.

Regarding the test itself, we feel that the SDK content tests are where any source-build differences in the xml doc should be caught.  It is the product's responsibility to tests that it creates the appropriate xml doc files.

**Comment by MichaelSimons** [comment]
The XmlDocTests were already removed in 7.0 - see https://github.com/dotnet/source-build/issues/3031.  This change should have been backported to 6.0 right away.
